cookiecutter/find.py

```diff
@@ -3,21 +3,28 @@
 import os
 from pathlib import Path
 
+from jinja2 import Environment
+
 from cookiecutter.exceptions import NonTemplatedInputDirException
 
 logger = logging.getLogger(__name__)
 
 
-def find_template(repo_dir: "os.PathLike[str]") -> Path:
+def find_template(repo_dir: "os.PathLike[str]", environment: Environment) -> Path:
     """Determine which child directory of ``repo_dir`` is the project template.
 
     :param repo_dir: Local directory of newly cloned repo.
+    :param environment: The jinja2 environment.
     :return: Relative path to project template.
     """
-    logger.debug('Searching %s for the project template.', repo_dir)
+    logger.debug("Searching %s for the project template.", repo_dir)
 
     for str_path in os.listdir(repo_dir):
-        if 'cookiecutter' in str_path and '{{' in str_path and '}}' in str_path:
+        if (
+            'cookiecutter' in str_path
+            and environment.variable_start_string in str_path
+            and environment.variable_end_string in str_path
```

Comment:
This, and the similar change in `generate.py`, are the crucial elements of this bugfix. Nice!